### PR TITLE
Implement group index

### PR DESCRIPTION
**Pull request by allenwq**
Replace #178

Depends on #274

### CODE REVIEW

**Comment by allenwq** [comment]
@lowjoel this is ready for review

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -77,4 +77,16 @@ def time_period_class(item)
     return 'currently-valid' if item.currently_valid?
     return 'expired' if item.expired?
   end
+
```
Comment (lowjoel) [inline app/helpers/application_helper.rb]: Move this to its own helper.

Comment (allenwq) [inline app/helpers/application_helper.rb]: Do you think announcement/achievements/assessments/etc index should use these helpers as well ?

**Comment by lowjoel** [comment]
Does this replace #178?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (lowjoel) [inline app/helpers/application_helper.rb]: Yes, they probably should.

Problem is that sometimes you'd rather the full button vs the small button.

Comment (allenwq) [inline app/helpers/application_helper.rb]: what do you mean by full button / small button? the size of button  ?

Comment (lowjoel) [inline app/helpers/application_helper.rb]: The one with the glyph inside, and the big button with text inside

Comment (allenwq) [inline app/helpers/application_helper.rb]: I see, but shall we use only small icons for index action ? like current coursemology does. What's your suggestion  ?

Comment (lowjoel) [inline app/helpers/application_helper.rb]: You might want to see how to integrate with Simple Form.

Big buttons for normal forms. Small buttons only for table/summary views.

**Comment by lowjoel** [comment]
Remember #274.

**Comment by allenwq** [comment]
@lowjoel  I also rebased here, but seems like coverall is died again ..

File: app/views/course/groups/_group.html.slim [hunk 2]
```diff
@@ -0,0 +1,6 @@
+tr
```
Comment (lowjoel) [inline app/views/course/groups/_group.html.slim]: Follow the page style of the admin announcements. @dariusf

File: config/locales/en/course/groups.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+en:
+  course:
+    groups:
+      index:
+        header: 'Groups'
+        new: 'New'
+        name: 'Name'
+        number_of_students: 'Number Of Students'
```
Comment (lowjoel) [inline config/locales/en/course/groups.yml]: "of". In English we do not capitalise the O in a conjunction.

Comment (allenwq) [inline config/locales/en/course/groups.yml]: Thanks !

**Comment by allenwq** [comment]
@lowjoel done

File: app/views/course/groups/index.html.slim [hunk 4]
```diff
@@ -0,0 +1,12 @@
+h1 = t('.header')
+
+= link_to t('.new'), new_course_group_path(current_course), class: 'btn btn-primary'
+
+table.table
+  thead
+    th = t('.name')
```
Comment (lowjoel) [inline app/views/course/groups/index.html.slim]: `common.name`

**Pull request merged**